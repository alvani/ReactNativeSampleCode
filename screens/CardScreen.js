import React from 'react';
import { 
  StyleSheet, 
  TouchableHighlight, 
  Text
} from 'react-native'
import { Card } from 'react-native-elements'
import ListScreen from './ListScreen'

export default class CardScreen extends ListScreen {
    formatContent(val) {
      if (val) {
        let idx = val.indexOf('[+');
        if (idx >= 0) {
          val = val.substring(0, idx);
        }
      }
      return val;
    }    

    renderItem = ({item}) => (
      <TouchableHighlight onPress={() => this.props.navigation.navigate('Detail', {uri: item.url})}>
      <Card                                    
        image={item.urlToImage ? { uri: item.urlToImage } : null}                
      >
        <Text style={styles.title}>
          {item.title} 
        </Text>
        <Text style={styles.dateSource}>
          {item.source.name + '  |  ' + this.formatTime(item.publishedAt)} 
        </Text>
        <Text style={styles.content}>
          {this.formatContent(item.content)} 
        </Text>        
      </Card>
      </TouchableHighlight>
    )
}

const styles = StyleSheet.create({
  title: {
    fontWeight: 'bold', 
    fontSize: 20, 
    marginBottom: 5
  },  
  dateSource: {
    fontWeight: 'bold'
  },
  content: {
    textAlign: 'justify'
  }
});