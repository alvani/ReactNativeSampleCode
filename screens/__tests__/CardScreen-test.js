import 'react-native';
import React from 'react';
import CardScreen from '../CardScreen'
import renderer from 'react-test-renderer';

jest.mock('../../Axios');

it('renders correctly', () => {
    const tree = renderer.create(<CardScreen></CardScreen>).toJSON();
    expect(tree).toMatchSnapshot();    
});

it('formats content correctly', () => {
    let ls = renderer.create(<CardScreen></CardScreen>);
    let val = 'ABCDEF[+';
    expect(ls.getInstance().formatContent(val)).toEqual('ABCDEF');        
});