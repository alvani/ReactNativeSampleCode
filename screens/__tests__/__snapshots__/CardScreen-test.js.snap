// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "height": 50,
        "marginTop": 25,
      }
    }
  >
    <RCTPicker
      items={
        Array [
          Object {
            "label": "Berita Teratas",
            "textColor": undefined,
            "value": "",
          },
          Object {
            "label": "Bisnis",
            "textColor": undefined,
            "value": "business",
          },
          Object {
            "label": "Hiburan",
            "textColor": undefined,
            "value": "entertainment",
          },
          Object {
            "label": "Kesehatan",
            "textColor": undefined,
            "value": "health",
          },
          Object {
            "label": "Sains",
            "textColor": undefined,
            "value": "science",
          },
          Object {
            "label": "Olah Raga",
            "textColor": undefined,
            "value": "sports",
          },
          Object {
            "label": "Teknologi",
            "textColor": undefined,
            "value": "technology",
          },
        ]
      }
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      selectedIndex={0}
      style={
        Array [
          Object {
            "height": 216,
          },
          undefined,
        ]
      }
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "lightgrey",
        "height": 1,
      }
    }
  />
</View>
`;
